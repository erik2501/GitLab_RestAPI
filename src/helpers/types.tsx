import { NamedExoticComponent } from "react";

export type Commit = {
    id: string;
    short_id: string;
    created_at: string;
    parent_ids: string[];
    title: string;
    message: string;
    author_name: string;
    author_email: string;
    authored_date: string;
    committer_name: string;
    committer_email: string;
    committed_date: string;
    trailers: unknown;
    web_url: string;
}

type Author = {
    id: number,
    username: string,
    name: string,
    state: string,
    avatar_url: string,
    web_url: string
}

export type User = {
    id: number;
    username: string;
    name: string;
    state: string;
}


export type Issue = {
    id: number,
    iid: number,
    project_id: number,
    title: string,
    description: string, 
    state: string,
    created_at: string,
    updated_at: string,
    closed_at: string, 
    closed_by: User,
    labels: string[], 
    milestone: string, 
    assignees: string[], 
    author: Author,
    type: string,
    assignee: string, 
    user_notes_count: number
}

type Namespace = {
    name: string
}

export type Project = {
    id: number,
    description: string,
    name: string,
    avatar_url: string //"https://gitlab.stud.idi.ntnu.no/uploads/-/system/project/avatar/17430/pingvin.png"
    namespace: Namespace
}

export type LoginInfo = {
    projectID: string | null,
    token: string | null
}