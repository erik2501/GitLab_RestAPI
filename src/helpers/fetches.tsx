import { Commit } from "./types";


export async function getCommits() {
    const response = await fetch("https://gitlab.stud.idi.ntnu.no/api/v4/projects/17430/repository/commits", {
        headers:
        {
            Authorization: "Bearer glpat-n3y-kCt83mAmv5KK63js"
        }
    })
    if (response.ok) {
        return await response.json() as Commit[];
    } else {
        console.log('something went wrong')
    }
}


