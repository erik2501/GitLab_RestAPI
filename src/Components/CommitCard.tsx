import { Commit } from '../helpers/types';


export default function CommitCard({ commit }: { commit: Commit }) {

    return (
        <div className='commitCard'>
            <h5>{commit.committed_date.toString()}</h5>
            <h2>{commit.title}</h2>
            <h5>{commit.committer_name}</h5>
            <h4>{commit.message}</h4>
        </div>
    );
}