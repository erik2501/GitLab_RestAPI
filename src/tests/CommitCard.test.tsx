import CommitCard from '../Components/CommitCard';
import { getCommits } from '../helpers/fetches';
import { Commit } from '../helpers/types';
import { useEffect, useState } from 'react';
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';


const dummyCommit: Commit = {
    id: '1',
    short_id: '2',
    created_at: '3',
    parent_ids: [''],
    title: 'title',
    message: 'message',
    author_name: 'name',
    author_email: 'epost',
    authored_date: "2021-05-02T18:02:30.000-05:00",
    committer_name: 'com_name',
    committer_email: 'com_email',
    committed_date: "2021-05-02T18:02:30.000-05:00",
    trailers: '',
    web_url: 'url',
};


it('renders when passed a commit', () => {
    const tree = renderer.create((<CommitCard commit={dummyCommit} />)).toJSON();
    expect(tree).toMatchSnapshot();
});

