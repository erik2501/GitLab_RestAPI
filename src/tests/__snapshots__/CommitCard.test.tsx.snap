// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when passed a commit 1`] = `
<div
  className="commitCard"
>
  <h5>
    2021-05-02T18:02:30.000-05:00
  </h5>
  <h2>
    title
  </h2>
  <h5>
    com_name
  </h5>
  <h4>
    message
  </h4>
</div>
`;
